src/Evolution/Systems/GeneralizedHarmonic/Constraints.cpp

```diff
@@ -461,15 +463,17 @@ void constraint_energy(
     double gauge_constraint_multiplier, double two_index_constraint_multiplier,
     double three_index_constraint_multiplier,
     double four_index_constraint_multiplier) noexcept {
-  for (auto& component : *energy) {
-    component = 0.0;
+  if (get_size(get(*energy)) != get_size(get<0>(f_constraint))) {
+    Scalar<DataType> energy_(get<0>(f_constraint));
```

Comment:
Sorry, I've referred you to a piece of code that didn't seem to have implemented all the changes I wanted yet. it is preferable to do this in a single line:
`*energy = Scalar<DataType>{get<0>(f_constraint)};`